product_contract/models/product_template.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2017 LasLabs Inc.
+# Copyright 2018 ACSONE SA/NV.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    is_contract = fields.Boolean("Is a contract")
+    property_contract_template_id = fields.Many2one(
+        comodel_name="contract.template",
+        string="Contract Template",
+        company_dependent=True,
+    )
+    default_qty = fields.Integer(string="Default Quantity", default=1)
+    recurring_rule_type = fields.Selection(
```

Comment:
FYI, I had done this once upon a time:

https://github.com/OCA/server-ux/pull/510

But no success :sweat: